<!doctype html>
<html lang="en">

<head>
	<meta charset="utf-8" />
	<link rel="apple-touch-icon" sizes="180x180" href="./apple-touch-icon.png">
	<link rel="icon" type="image/png" sizes="32x32" href="./favicon-32x32.png">
	<link rel="icon" type="image/png" sizes="16x16" href="./favicon-16x16.png">
	<link rel="icon" type="image/png" href="./favicon-16x16.png" />
	<link rel="manifest" href="./site.webmanifest">
	<link rel="mask-icon" href="./safari-pinned-tab.svg" color="#5bbad5">
	<meta name="msapplication-TileColor" content="#da532c">
	<meta name="theme-color" content="#ffffff">
	<meta name="viewport" content="width=device-width, initial-scale=1" />
	
		<link href="./_app/immutable/assets/5.990194d5.css" rel="stylesheet">
		<link href="./_app/immutable/assets/app.c6a029fe.css" rel="stylesheet">
		<link rel="modulepreload" href="./_app/immutable/entry/start.008c3a23.js">
		<link rel="modulepreload" href="./_app/immutable/chunks/scheduler.cd38125d.js">
		<link rel="modulepreload" href="./_app/immutable/chunks/singletons.86de3700.js">
		<link rel="modulepreload" href="./_app/immutable/entry/app.6f9145e1.js">
		<link rel="modulepreload" href="./_app/immutable/chunks/preload-helper.a4192956.js">
		<link rel="modulepreload" href="./_app/immutable/chunks/index.a6a2bf8f.js">
		<link rel="modulepreload" href="./_app/immutable/nodes/0.1790fe3c.js">
		<link rel="modulepreload" href="./_app/immutable/nodes/5.b8ce8349.js">
		<link rel="modulepreload" href="./_app/immutable/chunks/stores.6e73a4c1.js">
		<link rel="modulepreload" href="./_app/immutable/chunks/stores.3ed8a801.js"><title>GOAT: Geneset Ordinal Association Test - online geneset enrichment analyses</title><!-- HEAD_svelte-1hc78a4_START --><meta name="description" content="Geneset enrichment analysis for Geneset Ontology (GO) or KEGG pathways using the GOAT algorithm webtool. Online data analysis for your preranked genelist from e.g. proteomics or bulk/scRNAseq gene expression studies"><!-- HEAD_svelte-1hc78a4_END -->
</head>

<body data-sveltekit-preload-data="hover">
	<div style="display: contents">   <nav><div style="padding: 4px; margin-left: 20px;"><a href="/"><img src="/android-chrome-192x192.png" width="40" height="40" alt="GOAT"></a></div> <a href="/" style="margin-left: 5px;">Home</a> <a href="/goat" style="margin-left: 40px;" class="active">GOAT online</a> <a href="/genemap" style="margin-left: 40px;">gene ID mapping</a> <a href="/docs" style="margin-left: 40px;">Documentation</a></nav> <div tabindex="-1" role="button" aria-pressed="false"><div class="divH1" data-svelte-h="svelte-j0dsam"><h1>Online geneset enrichment analysis tool for the GOAT algorithm</h1> <div style="margin: 0px 150px 0px 150px; text-align: center;"><h3>Identify enriched genesets using GOAT and inspect your results using interactive tables and
				data visualizations.</h3> <img src="beta.png" width="200px" alt="BETA VERSION"></div></div> <div class="containerSettings"><div class="divH2" data-svelte-h="svelte-7havjb"><h2>Genelist</h2></div> <div class="divH3" data-svelte-h="svelte-31gbox"><h3>Load your input data; a table of genes and their p-values/effectsizes.</h3></div> <div class="pnlSettings"><table><tbody><tr><td style="vertical-align: top; width: 300px;"><input id="genelistFileDialog" type="file" accept=".csv,.tsv,.xlsx" multiple="false" style="display: none;"> <div id="genelist_dropzone" tabindex="0" role="button" aria-pressed="false"><p><i data-svelte-h="svelte-uobxa9">drag&amp;drop a genelist file or click to open a file dialog</i> <br> <i class="dropicon"></i> </p></div> <div style=""><br> <br> <b data-svelte-h="svelte-jrlymk">status:</b> <br> <div class="alert" data-svelte-h="svelte-1x7jbfu">no genelist loaded yet</div></div></td> <td><div class="helpSettings">Importantly, your input genelist / gene table must be prepared in a format that is
								compatible with this tool. In brief,
								<ul data-svelte-h="svelte-9uig0e"><li>File format: either CSV, TSV or Excel (.xlsx file, data on the first sheet)</li> <li>Required columns (column names must match exactly)</li> <ol><li><b>gene</b>: Human Entrez (NCBI) gene IDs</li> <li><b>symbol</b>: gene symbol</li> <li><b>effectsize</b>: effectsize or log2 foldchange</li> <li><b>pvalue</b>: gene p-value (preferably un-adjusted)</li> <li><b>signif</b>: was the adjusted p-value significant?
											<br>(boolean values, i.e. &#39;true&#39; or &#39;false&#39;)</li></ol> <li>No missing/empty values in any of these columns</li></ul>

								Further information and examples are available in the
								<a href="/docs" target="_blank" tabindex="-1">documentation</a>.

								<div style="margin-top: 20px;"> <button class="hrefUIaction" style="font-weight:700" >click here to load example genelist.</button>
									Instead of preparing and uploading your own genelist, you can immediately try this
									tool using an example genelist from a proteomics study on Alzheimer&#39;s disease by Wingo
									et al. (download link and reference available through the documentation section)</div></div></td></tr></tbody></table></div></div> <div class="containerSettings"><div class="divH2" data-svelte-h="svelte-1clroec"><h2>Genesets</h2></div> <div class="divH3" data-svelte-h="svelte-avc7bi"><h3>Select the genesets that should be used for GOAT analyses.</h3></div> <div class="pnlSettings"><div><table><tbody><tr><td style="display: flex; width: 180px;"><button class="btn btn-outline-blue" style="margin-right: 5px;" >Gene Ontology</button> </td> <td data-svelte-h="svelte-mcho4g"> </td> <td class="helpSettings" data-svelte-h="svelte-fflmn8">load GO genesets (typical use of this tool)</td></tr> <tr><td style="display: flex; width: 180px;"><button class="btn btn-outline-blue" style="margin-right: 5px;" >SynGO</button> </td> <td data-svelte-h="svelte-mcho4g"> </td> <td class="helpSettings" data-svelte-h="svelte-142skxt">load genesets from the synaptic knowledgebase (SynGO)</td></tr> <tr><td style="display: flex; width: 180px;"><input id="genesetGMTFileDialog" type="file" accept=".gmt" multiple="false" style="display: none;"> <button class="btn btn-outline-blue" style="margin-right: 5px;" >upload GMT file</button> </td> <td data-svelte-h="svelte-mcho4g"> </td> <td class="helpSettings" data-svelte-h="svelte-1deotz4">load genesets prepared in the GMT file format (i.e. load a GMT file from your
								computer)</td></tr></tbody></table></div> <div><br> <b data-svelte-h="svelte-jrlymk">status:</b> <br> <div class="alert" data-svelte-h="svelte-iyo4kz">no genesets loaded yet</div></div></div></div> <div class="containerSettings"><div class="divH2" data-svelte-h="svelte-1kt4qpn"><h2>GOAT analysis</h2></div> <div class="divH3" data-svelte-h="svelte-379imc"><h3>Settings &amp; apply algorithm.</h3></div> <div class="pnlSettings"><table><tbody><tr><td class="textRight" data-svelte-h="svelte-sanjpb">gene score type:</td> <td><select></select></td> <td class="helpSettings">recommended to set to &quot;effectsize&quot;, if available in your genelist
							<br>(refer to the
							<a href="/docs" target="_blank" tabindex="-1">Documentation</a>
							for preparing your input data accordingly)</td></tr> <tr><td class="textRight" data-svelte-h="svelte-7lt6jw">geneset size, minimum:</td> <td><input class="inputNumber" value="10"></td> <td class="helpSettings" data-svelte-h="svelte-10995iv">must be at least 10, recommended to leave at default (10)</td></tr> <tr><td class="textRight" data-svelte-h="svelte-aoup1q">geneset size, maximum:</td> <td><input class="inputNumber" value="1500"></td> <td class="helpSettings" data-svelte-h="svelte-1ol7e05">threshold on geneset size, i.e. intersection of the input geneset and your genelist.
							Typically set to 1500, recommended to leave at default. The largest allowed geneset
							size is [50% of the genelist], GOAT will automatically discard larger
							genesets</td></tr> <tr><td class="textRight valign-top" data-svelte-h="svelte-1qa4fvu">p-value adjustment:</td> <td colspan="2"><div><input type="radio" id="padjBonferroni" name="radioPadjust" checked> <label for="padjBonferroni" data-svelte-h="svelte-3hf4fz">Bonferroni</label></div> <div><input type="radio" id="padjFDR" name="radioPadjust" > <label for="padjFDR" data-svelte-h="svelte-auc0r">FDR</label></div></td></tr> <tr><td class="textRight" data-svelte-h="svelte-i0758a">adj. p-value threshold:</td> <td><input class="inputNumber" value="0.05"></td> <td class="helpSettings" data-svelte-h="svelte-gag5uy">genesets with adjusted p-value ≤ this threshold are considered significant. Must be
							a numeric value between 0 and 1</td></tr> <tr><td data-svelte-h="svelte-mcho4g"> </td> <td style="display: flex; width: 100px;"><button class="btn btn-action" style="margin-right: 5px;" disabled>START</button> </td> <td class="helpSettings"><div class="alert" data-svelte-h="svelte-1m62rtu">cannot start until the genelist and genesets are loaded</div></td></tr></tbody></table> </div></div> </div> <canvas id="mycanvas"></canvas> 
			
			<script>
				{
					__sveltekit_cgr8t8 = {
						base: new URL(".", location).pathname.slice(0, -1),
						env: {}
					};

					const element = document.currentScript.parentElement;

					const data = [null,null];

					Promise.all([
						import("./_app/immutable/entry/start.008c3a23.js"),
						import("./_app/immutable/entry/app.6f9145e1.js")
					]).then(([kit, app]) => {
						kit.start(app, element, {
							node_ids: [0, 5],
							data,
							form: null,
							error: null
						});
					});
				}
			</script>
		</div>
</body>

</html>