<!doctype html>
<html lang="en">

<head>
	<meta charset="utf-8" />
	<link rel="apple-touch-icon" sizes="180x180" href="./apple-touch-icon.png">
	<link rel="icon" type="image/png" sizes="32x32" href="./favicon-32x32.png">
	<link rel="icon" type="image/png" sizes="16x16" href="./favicon-16x16.png">
	<link rel="icon" type="image/png" href="./favicon-16x16.png" />
	<link rel="manifest" href="./site.webmanifest">
	<link rel="mask-icon" href="./safari-pinned-tab.svg" color="#5bbad5">
	<meta name="msapplication-TileColor" content="#da532c">
	<meta name="theme-color" content="#ffffff">
	<meta name="viewport" content="width=device-width, initial-scale=1" />
	
		<link href="./_app/immutable/assets/2.480caa0b.css" rel="stylesheet">
		<link href="./_app/immutable/assets/app.0f494d5b.css" rel="stylesheet">
		<link rel="modulepreload" href="./_app/immutable/entry/start.136301d3.js">
		<link rel="modulepreload" href="./_app/immutable/chunks/scheduler.a30ae394.js">
		<link rel="modulepreload" href="./_app/immutable/chunks/singletons.ed106ba1.js">
		<link rel="modulepreload" href="./_app/immutable/entry/app.242b7cfa.js">
		<link rel="modulepreload" href="./_app/immutable/chunks/index.49b3fccf.js">
		<link rel="modulepreload" href="./_app/immutable/nodes/0.9f3eb7bd.js">
		<link rel="modulepreload" href="./_app/immutable/nodes/2.4f417654.js">
		<link rel="modulepreload" href="./_app/immutable/chunks/stores.7fc2773d.js"><title>GOAT: Geneset Ordinal Association Test</title><!-- HEAD_svelte-1b313dc_START --><meta name="description" content="Geneset enrichment analysis for Geneset Ontology (GO) or KEGG pathways using the GOAT algorithm webtool. Online data analysis for your preranked genelist from e.g. proteomics or bulk/scRNAseq gene expression studies"><!-- HEAD_svelte-1b313dc_END -->
</head>

<body data-sveltekit-preload-data="hover">
	<div style="display: contents">   <nav><div style="padding: 4px; margin-left: 20px;"><a href="/" class="active"><img src="/android-chrome-192x192.png" width="40" height="40" alt="GOAT"></a></div> <a href="/" style="margin-left: 5px;" class="active">Home</a> <a href="/goat" style="margin-left: 40px;">GOAT online</a> <a href="/genemap" style="margin-left: 40px;">gene ID mapping</a> <a href="/docs" style="margin-left: 40px;">Documentation</a></nav> <div class="divTitle svelte-1vmxgqy" data-svelte-h="svelte-p77g5c"><h1 style="" class="svelte-1vmxgqy">GOAT online geneset enrichment analysis</h1> <div><h3 class="svelte-1vmxgqy">Identify enriched genesets in a preranked genelist generated by e.g. proteomics or gene
			expression studies using the GOAT algorithm.</h3></div></div> <div style="background-color: white; padding: 0px 50px 0px 50px;"><div style="padding-top: 50px" data-svelte-h="svelte-1qljcv1"><h2 class="svelte-1vmxgqy">The GOAT algorithm for geneset enrichment analysis</h2> <div><p>The GOAT algorithm has not been published yet but a preprint is available, please cite it
				when using the early-access version of GOAT;
				<br> <i>Koopmans, F. (2023). GOAT: efficient and robust identification of geneset enrichment.</i> <br> <a href="https://doi.org/10.1101/2023.12.10.570979" target="_blank" rel="nofollow">https://doi.org/10.1101/2023.12.10.570979</a></p></div> <div style="margin-top: 25px;">GOAT features:
			<ul class="svelte-1vmxgqy"><li class="svelte-1vmxgqy">Sensitive; more significant genesets as compared to other methods</li> <li class="svelte-1vmxgqy">Accurate; geneset p-values are accurate under the null hypothesis</li> <li class="svelte-1vmxgqy">Fast; completes in seconds</li> <li class="svelte-1vmxgqy">No arbitrary parameters (unlike ORA, no fiddle parameters for &quot;foreground genes&quot;)</li> <li class="svelte-1vmxgqy">Bootstrapping approach always uses appropriate &quot;background set&quot; (unlike ORA)</li> <li class="svelte-1vmxgqy">Available as a R package and through this website (<i>GOAT online</i> in above menu)</li></ul> <span><i>*ORA: overrepresentation analysis (i.e. classical Fisher-exact or hypergeometric test)</i></span></div></div> <div style="padding-top: 50px"><h2 class="svelte-1vmxgqy" data-svelte-h="svelte-10ytmrf">GOAT finds more significant genesets</h2> <div data-svelte-h="svelte-t7brcz">GOAT identifies more significant GO terms as compared to GSEA and ORA across 6 omics studies.</div> <div style="margin-top: 25px;"><img src="/barplot_signif_count.svg" width="800" height="450" alt="barplot significant geneset counts"></div> <p style="padding: 0px 50px 0px 20px;" data-svelte-h="svelte-8x3x4l">Differential expression analysis results from each study (panels) were extracted from
			supplementary data and subjected to 5 approaches (colors) for identifying overrepresented GO
			terms. GO MF, GO CC and GO BP represent the respective Gene Ontology domains Molecular
			Functions, Cellular Components and Biological Processes. The x-axis shows the number of
			significant genesets identified by each method after Bonferroni adjustment at α = 0.05
			and adjustment for testing 3 sources of genesets.</p></div> <div style="padding-top: 50px"><h2 class="svelte-1vmxgqy" data-svelte-h="svelte-1cxefkq">GOAT p-values are accurate</h2> <div data-svelte-h="svelte-36zne0">In simulations that generate 200 thousand random genelists and genesets, from small to large,
			we find that geneset p-values estimated by GOAT are accurate under the null hypothesis.</div> <div style="margin-top: 25px;"><img src="/null_simulations.svg" width="800" height="225" alt="GOAT p-values are accurate under the null hypothesis"></div> <p style="padding: 0px 50px 0px 20px;" data-svelte-h="svelte-ogerj8">The y-axis shows the observed geneset p-value for a given p-value threshold on the x-axis
			(both on -log10 scale, p-values are shown as-is without multiple testing correction). For
			example, selecting a threshold of p = 0.01 (2 on -log10 scale) on the x-axis shows the
			proportion of observed genesets with p ≤ 0.01 on the y-axis. Since randomized genesets were
			used in these simulations, expected values are on the diagonal (dashed line).</p></div> <div style="margin-top: 100px"><h2 class="svelte-1vmxgqy" data-svelte-h="svelte-69l058">Description of the GOAT algorithm</h2> <p data-svelte-h="svelte-150sgbv">In brief, the Geneset Ordinal Association Test (GOAT) is a parameter-free permutation-based
			algorithm for geneset enrichment analysis. It is easy to use via the online webtool or R
			package, computationally efficient and the resulting geneset p-values are well calibrated
			under the null hypothesis and invariant to geneset size. Application to various real-world
			proteomics and gene expression studies demonstrates that GOAT consistently identifies more
			significant Gene Ontology terms as compared to alternative methods.</p> <div><img src="/goat_algorithm.png" alt="GOAT algorithm"></div> <div style="padding: 0px 50px 0px 10px;" data-svelte-h="svelte-rfve2k"><ol><li class="svelte-1vmxgqy">Required input are a list of genes and their respective test statistics (p-value /
					effectsize), and a list of genesets obtained from GO or alternative resources.</li> <li class="svelte-1vmxgqy">Test statistics from the genelist are transformed to gene scores by rank(-pvalue)^2 or
					rank(effectsize)^2 depending on user-input, i.e. smaller p-values translate to higher gene
					scores. The result is a skewed gene score distribution.</li> <li class="svelte-1vmxgqy">For each geneset size N (number of genes), bootstrapping procedures generate a null
					distribution of geneset scores. This yields a skewed-normal distribution for small
					genesets and converges to a normal distribution for large genesets.</li> <li class="svelte-1vmxgqy">Geneset significance is determined for each geneset by comparing its score (mean of
					respective gene scores) against a null distribution of the same size (N).</li></ol></div></div> </div> 
			
			<script>
				{
					__sveltekit_snvp3l = {
						base: new URL(".", location).pathname.slice(0, -1),
						env: {}
					};

					const element = document.currentScript.parentElement;

					const data = [null,null];

					Promise.all([
						import("./_app/immutable/entry/start.136301d3.js"),
						import("./_app/immutable/entry/app.242b7cfa.js")
					]).then(([kit, app]) => {
						kit.start(app, element, {
							node_ids: [0, 2],
							data,
							form: null,
							error: null
						});
					});
				}
			</script>
		</div>
</body>

</html>