import{s as Ke,f as t,a as i,l as Y,A as Ue,g as s,d as n,c as r,h,m as Z,B as m,j as a,C as ee,k as o,x as e,i as W,y as be,z as We}from"../chunks/scheduler.cd38125d.js";import{S as Xe,i as Je}from"../chunks/index.a6a2bf8f.js";/* empty css                    */import{p as Qe}from"../chunks/stores.f08d5393.js";function Ye(g){let y,L,l,f,_,G,Ge,te,A,se,ae,E,ne,ie,k,re,le,V,oe,J,O,Ae=`<h1 style="" class="svelte-1vmxgqy">GOAT online geneset enrichment analysis</h1> <div><h3 class="svelte-1vmxgqy">Identify enriched genesets in a preranked genelist generated by e.g. proteomics or gene
			expression studies using the GOAT algorithm.</h3></div>`,Q,c,M,Oe=`<h2 class="svelte-1vmxgqy">The GOAT algorithm for geneset enrichment analysis</h2> <div><p>The GOAT algorithm has not been published yet but a preprint is available, please cite it
				when using the early-access version of GOAT;
				<br/> <i>Koopmans, F. (2023). GOAT: efficient and robust identification of geneset enrichment.</i> <br/> <a href="https://doi.org/10.1101/2023.12.10.570979" target="_blank" rel="nofollow">https://doi.org/10.1101/2023.12.10.570979</a></p></div> <div style="margin-top: 25px;">GOAT features:
			<ul class="svelte-1vmxgqy"><li class="svelte-1vmxgqy">Sensitive; more significant genesets as compared to other methods</li> <li class="svelte-1vmxgqy">Accurate; geneset p-values are accurate under the null hypothesis</li> <li class="svelte-1vmxgqy">Fast; completes in seconds</li> <li class="svelte-1vmxgqy">No arbitrary parameters (unlike ORA, no fiddle parameters for &quot;foreground genes&quot;)</li> <li class="svelte-1vmxgqy">Bootstrapping approach always uses appropriate &quot;background set&quot; (unlike ORA)</li> <li class="svelte-1vmxgqy">Available as a R package and through this website (<i>GOAT online</i> in above menu)</li></ul> <span><i>*ORA: overrepresentation analysis (i.e. classical Fisher-exact or hypergeometric test)</i></span></div>`,de,p,H,Te="GOAT finds more significant genesets",ce,P,Ce="GOAT identifies more significant GO terms as compared to GSEA and ORA across 6 omics studies.",he,N,T,qe,ge,z,we=`Differential expression analysis results from each study (panels) were extracted from
			supplementary data and subjected to 5 approaches (colors) for identifying overrepresented GO
			terms. GO MF, GO CC and GO BP represent the respective Gene Ontology domains Molecular
			Functions, Cellular Components and Biological Processes. The x-axis shows the number of
			significant genesets identified by each method after Bonferroni adjustment at α = 0.05
			and adjustment for testing 3 sources of genesets.`,pe,u,F,Ie="GOAT p-values are accurate",ue,S,De=`In simulations that generate 200 thousand random genelists and genesets, from small to large,
			we find that geneset p-values estimated by GOAT are accurate under the null hypothesis.`,ve,$,C,Ee,me,R,ke=`The y-axis shows the observed geneset p-value for a given p-value threshold on the x-axis
			(both on -log10 scale, p-values are shown as-is without multiple testing correction). For
			example, selecting a threshold of p = 0.01 (2 on -log10 scale) on the x-axis shows the
			proportion of observed genesets with p ≤ 0.01 on the y-axis. Since randomized genesets were
			used in these simulations, expected values are on the diagonal (dashed line).`,fe,v,j,Ve="Description of the GOAT algorithm",xe,K,Me=`In brief, the Geneset Ordinal Association Test (GOAT) is a parameter-free permutation-based
			algorithm for geneset enrichment analysis. It is easy to use via the online webtool or R
			package, computationally efficient and the resulting geneset p-values are well calibrated
			under the null hypothesis and invariant to geneset size. Application to various real-world
			proteomics and gene expression studies demonstrates that GOAT consistently identifies more
			significant Gene Ontology terms as compared to alternative methods.`,ye,X,U,He,_e,B,ze=`<ol><li class="svelte-1vmxgqy">Required input are a list of genes and their respective test statistics (p-value /
					effectsize), and a list of genesets obtained from GO or alternative resources.</li> <li class="svelte-1vmxgqy">Test statistics from the genelist are transformed to gene scores by rank(-pvalue)^2 or
					rank(effectsize)^2 depending on user-input, i.e. smaller p-values translate to higher gene
					scores. The result is a skewed gene score distribution.</li> <li class="svelte-1vmxgqy">For each geneset size N (number of genes), bootstrapping procedures generate a null
					distribution of geneset scores. This yields a skewed-normal distribution for small
					genesets and converges to a normal distribution for large genesets.</li> <li class="svelte-1vmxgqy">Geneset significance is determined for each geneset by comparing its score (mean of
					respective gene scores) against a null distribution of the same size (N).</li></ol>`;return{c(){y=t("meta"),L=i(),l=t("nav"),f=t("div"),_=t("a"),G=t("img"),te=i(),A=t("a"),se=Y("Home"),ae=i(),E=t("a"),ne=Y("GOAT online"),ie=i(),k=t("a"),re=Y("gene ID mapping"),le=i(),V=t("a"),oe=Y("Documentation"),J=i(),O=t("div"),O.innerHTML=Ae,Q=i(),c=t("div"),M=t("div"),M.innerHTML=Oe,de=i(),p=t("div"),H=t("h2"),H.textContent=Te,ce=i(),P=t("div"),P.textContent=Ce,he=i(),N=t("div"),T=t("img"),ge=i(),z=t("p"),z.textContent=we,pe=i(),u=t("div"),F=t("h2"),F.textContent=Ie,ue=i(),S=t("div"),S.textContent=De,ve=i(),$=t("div"),C=t("img"),me=i(),R=t("p"),R.textContent=ke,fe=i(),v=t("div"),j=t("h2"),j.textContent=Ve,xe=i(),K=t("p"),K.textContent=Me,ye=i(),X=t("div"),U=t("img"),_e=i(),B=t("div"),B.innerHTML=ze,this.h()},l(d){const b=Ue("svelte-1b313dc",document.head);y=s(b,"META",{name:!0,content:!0}),b.forEach(n),L=r(d),l=s(d,"NAV",{});var x=h(l);f=s(x,"DIV",{style:!0});var Fe=h(f);_=s(Fe,"A",{href:!0,class:!0});var Re=h(_);G=s(Re,"IMG",{src:!0,width:!0,height:!0,alt:!0}),Re.forEach(n),Fe.forEach(n),te=r(x),A=s(x,"A",{href:!0,style:!0,class:!0});var je=h(A);se=Z(je,"Home"),je.forEach(n),ae=r(x),E=s(x,"A",{href:!0,style:!0});var Be=h(E);ne=Z(Be,"GOAT online"),Be.forEach(n),ie=r(x),k=s(x,"A",{href:!0,style:!0});var Le=h(k);re=Z(Le,"gene ID mapping"),Le.forEach(n),le=r(x),V=s(x,"A",{href:!0,style:!0});var Pe=h(V);oe=Z(Pe,"Documentation"),Pe.forEach(n),x.forEach(n),J=r(d),O=s(d,"DIV",{class:!0,"data-svelte-h":!0}),m(O)!=="svelte-p77g5c"&&(O.innerHTML=Ae),Q=r(d),c=s(d,"DIV",{style:!0});var q=h(c);M=s(q,"DIV",{style:!0,"data-svelte-h":!0}),m(M)!=="svelte-1qljcv1"&&(M.innerHTML=Oe),de=r(q),p=s(q,"DIV",{style:!0});var w=h(p);H=s(w,"H2",{class:!0,"data-svelte-h":!0}),m(H)!=="svelte-10ytmrf"&&(H.textContent=Te),ce=r(w),P=s(w,"DIV",{"data-svelte-h":!0}),m(P)!=="svelte-t7brcz"&&(P.textContent=Ce),he=r(w),N=s(w,"DIV",{style:!0});var Ne=h(N);T=s(Ne,"IMG",{src:!0,width:!0,height:!0,alt:!0}),Ne.forEach(n),ge=r(w),z=s(w,"P",{style:!0,"data-svelte-h":!0}),m(z)!=="svelte-8x3x4l"&&(z.textContent=we),w.forEach(n),pe=r(q),u=s(q,"DIV",{style:!0});var I=h(u);F=s(I,"H2",{class:!0,"data-svelte-h":!0}),m(F)!=="svelte-1cxefkq"&&(F.textContent=Ie),ue=r(I),S=s(I,"DIV",{"data-svelte-h":!0}),m(S)!=="svelte-36zne0"&&(S.textContent=De),ve=r(I),$=s(I,"DIV",{style:!0});var Se=h($);C=s(Se,"IMG",{src:!0,width:!0,height:!0,alt:!0}),Se.forEach(n),me=r(I),R=s(I,"P",{style:!0,"data-svelte-h":!0}),m(R)!=="svelte-ogerj8"&&(R.textContent=ke),I.forEach(n),fe=r(q),v=s(q,"DIV",{style:!0});var D=h(v);j=s(D,"H2",{class:!0,"data-svelte-h":!0}),m(j)!=="svelte-69l058"&&(j.textContent=Ve),xe=r(D),K=s(D,"P",{"data-svelte-h":!0}),m(K)!=="svelte-150sgbv"&&(K.textContent=Me),ye=r(D),X=s(D,"DIV",{});var $e=h(X);U=s($e,"IMG",{src:!0,alt:!0}),$e.forEach(n),_e=r(D),B=s(D,"DIV",{style:!0,"data-svelte-h":!0}),m(B)!=="svelte-rfve2k"&&(B.innerHTML=ze),D.forEach(n),q.forEach(n),this.h()},h(){document.title="GOAT: Geneset Ordinal Association Test",a(y,"name","description"),a(y,"content","Geneset enrichment analysis for Geneset Ontology (GO) or KEGG pathways using the GOAT algorithm webtool. Online data analysis for your preranked genelist from e.g. proteomics or bulk/scRNAseq gene expression studies"),ee(G.src,Ge=g[0]+"android-chrome-192x192.png")||a(G,"src",Ge),a(G,"width","40"),a(G,"height","40"),a(G,"alt","GOAT"),a(_,"href",g[0]),a(_,"class","active"),o(f,"padding","4px"),o(f,"margin-left","20px"),a(A,"href",g[0]),o(A,"margin-left","5px"),a(A,"class","active"),a(E,"href",g[0]+"goat"),o(E,"margin-left","40px"),a(k,"href",g[0]+"genemap"),o(k,"margin-left","40px"),a(V,"href",g[0]+"docs"),o(V,"margin-left","40px"),a(O,"class","divTitle svelte-1vmxgqy"),o(M,"padding-top","50px"),a(H,"class","svelte-1vmxgqy"),ee(T.src,qe=g[0]+"barplot_signif_count.svg")||a(T,"src",qe),a(T,"width","800"),a(T,"height","450"),a(T,"alt","barplot significant geneset counts"),o(N,"margin-top","25px"),o(z,"padding","0px 50px 0px 20px"),o(p,"padding-top","50px"),a(F,"class","svelte-1vmxgqy"),ee(C.src,Ee=g[0]+"null_simulations.svg")||a(C,"src",Ee),a(C,"width","800"),a(C,"height","225"),a(C,"alt","GOAT p-values are accurate under the null hypothesis"),o($,"margin-top","25px"),o(R,"padding","0px 50px 0px 20px"),o(u,"padding-top","50px"),a(j,"class","svelte-1vmxgqy"),ee(U.src,He=g[0]+"goat_algorithm.png")||a(U,"src",He),a(U,"alt","GOAT algorithm"),o(B,"padding","0px 50px 0px 10px"),o(v,"margin-top","100px"),o(c,"background-color","white"),o(c,"padding","0px 50px 0px 50px")},m(d,b){e(document.head,y),W(d,L,b),W(d,l,b),e(l,f),e(f,_),e(_,G),e(l,te),e(l,A),e(A,se),e(l,ae),e(l,E),e(E,ne),e(l,ie),e(l,k),e(k,re),e(l,le),e(l,V),e(V,oe),W(d,J,b),W(d,O,b),W(d,Q,b),W(d,c,b),e(c,M),e(c,de),e(c,p),e(p,H),e(p,ce),e(p,P),e(p,he),e(p,N),e(N,T),e(p,ge),e(p,z),e(c,pe),e(c,u),e(u,F),e(u,ue),e(u,S),e(u,ve),e(u,$),e($,C),e(u,me),e(u,R),e(c,fe),e(c,v),e(v,j),e(v,xe),e(v,K),e(v,ye),e(v,X),e(X,U),e(v,_e),e(v,B)},p:be,i:be,o:be,d(d){d&&(n(L),n(l),n(J),n(O),n(Q),n(c)),n(y)}}}function Ze(g,y,L){let l;We(g,Qe,_=>L(1,l=_));const f=!!l&&l.url.hostname.endsWith("github.io")&&"/goat/"||"/";return console.log("base="+f),[f]}class nt extends Xe{constructor(y){super(),Je(this,y,Ze,Ye,Ke,{})}}export{nt as component};
