import{s as S,f as n,a as A,A as V,g as o,d as r,c as G,h as P,B as O,j as x,C as q,k as y,x as a,i as _,y as D,z as $}from"../chunks/scheduler.7144be00.js";import{S as K,i as J}from"../chunks/index.3da70c46.js";/* empty css                    */import{p as Q}from"../chunks/stores.475d77e9.js";import{e as b}from"../chunks/singletons.bd5e124a.js";function U(u){let l,T,e,s,d,w='<img src="android-chrome-192x192.png" width="40" height="40" alt="GOAT"/>',M,p,E="Home",I,g,F="GOAT online",R,c,H="gene ID mapping",z,h,L="Documentation",C,f,N=`<h1 style="" class="svelte-1vmxgqy">GOAT online geneset enrichment analysis</h1> <div><h3 class="svelte-1vmxgqy">Identify enriched genesets in a preranked genelist generated by e.g. proteomics or gene
			expression studies using the GOAT algorithm.</h3></div>`,k,m,j=`<div style="padding-top: 50px"><h2 class="svelte-1vmxgqy">The GOAT algorithm for geneset enrichment analysis</h2> <div><p>The GOAT algorithm has not been published yet but a preprint is available, please cite it
				when using the early-access version of GOAT;
				<br/> <i>Koopmans, F. (2023). GOAT: efficient and robust identification of geneset enrichment.</i> <br/> <a href="https://doi.org/10.1101/2023.12.10.570979" target="_blank" rel="nofollow">https://doi.org/10.1101/2023.12.10.570979</a></p></div> <div style="margin-top: 25px;">GOAT features:
			<ul class="svelte-1vmxgqy"><li class="svelte-1vmxgqy">Sensitive; more significant genesets as compared to other methods</li> <li class="svelte-1vmxgqy">Accurate; geneset p-values are accurate under the null hypothesis</li> <li class="svelte-1vmxgqy">Fast; completes in seconds</li> <li class="svelte-1vmxgqy">No arbitrary parameters (unlike ORA, no fiddle parameters for &quot;foreground genes&quot;)</li> <li class="svelte-1vmxgqy">Bootstrapping approach always uses appropriate &quot;background set&quot; (unlike ORA)</li> <li class="svelte-1vmxgqy">Available as a R package and through this website (<i>GOAT online</i> in above menu)</li></ul> <span><i>*ORA: overrepresentation analysis (i.e. classical Fisher-exact or hypergeometric test)</i></span></div></div> <div style="padding-top: 50px"><h2 class="svelte-1vmxgqy">GOAT finds more significant genesets</h2> <div>GOAT identifies more significant GO terms as compared to GSEA and ORA across 6 omics studies.</div> <div style="margin-top: 25px;"><img src="barplot_signif_count.svg" width="800" height="450" alt="barplot significant geneset counts"/></div> <p style="padding: 0px 50px 0px 20px;">Differential expression analysis results from each study (panels) were extracted from
			supplementary data and subjected to 5 approaches (colors) for identifying overrepresented GO
			terms. GO MF, GO CC and GO BP represent the respective Gene Ontology domains Molecular
			Functions, Cellular Components and Biological Processes. The x-axis shows the number of
			significant genesets identified by each method after Bonferroni adjustment at α = 0.05
			and adjustment for testing 3 sources of genesets.</p></div> <div style="padding-top: 50px"><h2 class="svelte-1vmxgqy">GOAT p-values are accurate</h2> <div>In simulations that generate 200 thousand random genelists and genesets, from small to large,
			we find that geneset p-values estimated by GOAT are accurate under the null hypothesis.</div> <div style="margin-top: 25px;"><img src="null_simulations.svg" width="800" height="225" alt="GOAT p-values are accurate under the null hypothesis"/></div> <p style="padding: 0px 50px 0px 20px;">The y-axis shows the observed geneset p-value for a given p-value threshold on the x-axis
			(both on -log10 scale, p-values are shown as-is without multiple testing correction). For
			example, selecting a threshold of p = 0.01 (2 on -log10 scale) on the x-axis shows the
			proportion of observed genesets with p ≤ 0.01 on the y-axis. Since randomized genesets were
			used in these simulations, expected values are on the diagonal (dashed line).</p></div> <div style="margin-top: 100px"><h2 class="svelte-1vmxgqy">Description of the GOAT algorithm</h2> <p>In brief, the Geneset Ordinal Association Test (GOAT) is a parameter-free permutation-based
			algorithm for geneset enrichment analysis. It is easy to use via the online webtool or R
			package, computationally efficient and the resulting geneset p-values are well calibrated
			under the null hypothesis and invariant to geneset size. Application to various real-world
			proteomics and gene expression studies demonstrates that GOAT consistently identifies more
			significant Gene Ontology terms as compared to alternative methods.</p> <div><img src="goat_algorithm.png" alt="GOAT algorithm"/></div> <div style="padding: 0px 50px 0px 10px;"><ol><li class="svelte-1vmxgqy">Required input are a list of genes and their respective test statistics (p-value /
					effectsize), and a list of genesets obtained from GO or alternative resources.</li> <li class="svelte-1vmxgqy">Test statistics from the genelist are transformed to gene scores by rank(-pvalue)^2 or
					rank(effectsize)^2 depending on user-input, i.e. smaller p-values translate to higher gene
					scores. The result is a skewed gene score distribution.</li> <li class="svelte-1vmxgqy">For each geneset size N (number of genes), bootstrapping procedures generate a null
					distribution of geneset scores. This yields a skewed-normal distribution for small
					genesets and converges to a normal distribution for large genesets.</li> <li class="svelte-1vmxgqy">Geneset significance is determined for each geneset by comparing its score (mean of
					respective gene scores) against a null distribution of the same size (N).</li></ol></div></div>`;return{c(){l=n("meta"),T=A(),e=n("nav"),s=n("div"),d=n("a"),d.innerHTML=w,M=A(),p=n("a"),p.textContent=E,I=A(),g=n("a"),g.textContent=F,R=A(),c=n("a"),c.textContent=H,z=A(),h=n("a"),h.textContent=L,C=A(),f=n("div"),f.innerHTML=N,k=A(),m=n("div"),m.innerHTML=j,this.h()},l(t){const v=V("svelte-1b313dc",document.head);l=o(v,"META",{name:!0,content:!0}),v.forEach(r),T=G(t),e=o(t,"NAV",{});var i=P(e);s=o(i,"DIV",{style:!0});var B=P(s);d=o(B,"A",{href:!0,"data-svelte-h":!0}),O(d)!=="svelte-119aptw"&&(d.innerHTML=w),B.forEach(r),M=G(i),p=o(i,"A",{href:!0,style:!0,"data-svelte-h":!0}),O(p)!=="svelte-xsa54j"&&(p.textContent=E),I=G(i),g=o(i,"A",{href:!0,style:!0,"data-svelte-h":!0}),O(g)!=="svelte-o349n8"&&(g.textContent=F),R=G(i),c=o(i,"A",{href:!0,style:!0,"data-svelte-h":!0}),O(c)!=="svelte-sn5cge"&&(c.textContent=H),z=G(i),h=o(i,"A",{href:!0,style:!0,"data-svelte-h":!0}),O(h)!=="svelte-1dikf5g"&&(h.textContent=L),i.forEach(r),C=G(t),f=o(t,"DIV",{class:!0,"data-svelte-h":!0}),O(f)!=="svelte-p77g5c"&&(f.innerHTML=N),k=G(t),m=o(t,"DIV",{style:!0,"data-svelte-h":!0}),O(m)!=="svelte-1y2idh4"&&(m.innerHTML=j),this.h()},h(){document.title="GOAT: Geneset Ordinal Association Test",x(l,"name","description"),x(l,"content","Geneset enrichment analysis for Geneset Ontology (GO) or KEGG pathways using the GOAT algorithm webtool. Online data analysis for your preranked genelist from e.g. proteomics or bulk/scRNAseq gene expression studies"),x(d,"href",b+"/"),q(d,"active",u[0]==="home"),y(s,"padding","4px"),y(s,"margin-left","20px"),x(p,"href",b+"/"),y(p,"margin-left","5px"),q(p,"active",u[0]==="home"),x(g,"href",b+"/goat"),y(g,"margin-left","40px"),q(g,"active",u[0]==="goat"),x(c,"href",b+"/genemap"),y(c,"margin-left","40px"),q(c,"active",u[0]==="genemap"),x(h,"href",b+"/docs"),y(h,"margin-left","40px"),q(h,"active",u[0]==="docs"),x(f,"class","divTitle svelte-1vmxgqy"),y(m,"background-color","white"),y(m,"padding","0px 50px 0px 50px")},m(t,v){a(document.head,l),_(t,T,v),_(t,e,v),a(e,s),a(s,d),a(e,M),a(e,p),a(e,I),a(e,g),a(e,R),a(e,c),a(e,z),a(e,h),_(t,C,v),_(t,f,v),_(t,k,v),_(t,m,v)},p:D,i:D,o:D,d(t){t&&(r(T),r(e),r(C),r(f),r(k),r(m)),r(l)}}}function W(u,l,T){let e;$(u,Q,w=>T(1,e=w));const s=e&&e.url?e.url.pathname.replace(".html",""):"/";return[s===b+"/goat"&&"goat"||s===b+"/docs"&&"docs"||s===b+"/genemap"&&"genemap"||"home"]}class se extends K{constructor(l){super(),J(this,l,W,U,S,{})}}export{se as component};
