import{s as P,f as g,a as G,A as V,g as c,d as h,c as O,h as S,B as T,j as b,C as v,k as A,x as a,i as q,y as j,z as $}from"../chunks/scheduler.7144be00.js";import{S as K,i as J}from"../chunks/index.3da70c46.js";import{i}from"../chunks/util.14f89131.js";import{p as Q}from"../chunks/stores.bbded9a9.js";function U(n){let m,_,t,u,f,R='<img src="./android-chrome-192x192.png" width="40" height="40" alt="GOAT"/>',k,o,D="Home",z,l,F="GOAT online",M,r,H="gene ID mapping",I,d,E="Documentation",w,y,L=`<h1 style="" class="svelte-1vmxgqy">GOAT online geneset enrichment analysis</h1> <div><h3 class="svelte-1vmxgqy">Identify enriched genesets in a preranked genelist generated by e.g. proteomics or gene
			expression studies using the GOAT algorithm.</h3></div>`,C,x,B=`<div style="padding-top: 50px"><h2 class="svelte-1vmxgqy">The GOAT algorithm for geneset enrichment analysis</h2> <div><p>The GOAT algorithm has not been published yet but a preprint is available, please cite it
				when using the early-access version of GOAT;
				<br/> <i>Koopmans, F. (2023). GOAT: efficient and robust identification of geneset enrichment.</i> <br/> <a href="https://doi.org/10.1101/2023.12.10.570979" target="_blank" rel="nofollow">https://doi.org/10.1101/2023.12.10.570979</a></p></div> <div style="margin-top: 25px;">GOAT features:
			<ul class="svelte-1vmxgqy"><li class="svelte-1vmxgqy">Sensitive; more significant genesets as compared to other methods</li> <li class="svelte-1vmxgqy">Accurate; geneset p-values are accurate under the null hypothesis</li> <li class="svelte-1vmxgqy">Fast; completes in seconds</li> <li class="svelte-1vmxgqy">No arbitrary parameters (unlike ORA, no fiddle parameters for &quot;foreground genes&quot;)</li> <li class="svelte-1vmxgqy">Bootstrapping approach always uses appropriate &quot;background set&quot; (unlike ORA)</li> <li class="svelte-1vmxgqy">Available as a R package and through this website (<i>GOAT online</i> in above menu)</li></ul> <span><i>*ORA: overrepresentation analysis (i.e. classical Fisher-exact or hypergeometric test)</i></span></div></div> <div style="padding-top: 50px"><h2 class="svelte-1vmxgqy">GOAT finds more significant genesets</h2> <div>GOAT identifies more significant GO terms as compared to GSEA and ORA across 6 omics studies.</div> <div style="margin-top: 25px;"><img src="barplot_signif_count.svg" width="800" height="450" alt="barplot significant geneset counts"/></div> <p style="padding: 0px 50px 0px 20px;">Differential expression analysis results from each study (panels) were extracted from
			supplementary data and subjected to 5 approaches (colors) for identifying overrepresented GO
			terms. GO MF, GO CC and GO BP represent the respective Gene Ontology domains Molecular
			Functions, Cellular Components and Biological Processes. The x-axis shows the number of
			significant genesets identified by each method after Bonferroni adjustment at α = 0.05
			and adjustment for testing 3 sources of genesets.</p></div> <div style="padding-top: 50px"><h2 class="svelte-1vmxgqy">GOAT p-values are accurate</h2> <div>In simulations that generate 200 thousand random genelists and genesets, from small to large,
			we find that geneset p-values estimated by GOAT are accurate under the null hypothesis.</div> <div style="margin-top: 25px;"><img src="null_simulations.svg" width="800" height="225" alt="GOAT p-values are accurate under the null hypothesis"/></div> <p style="padding: 0px 50px 0px 20px;">The y-axis shows the observed geneset p-value for a given p-value threshold on the x-axis
			(both on -log10 scale, p-values are shown as-is without multiple testing correction). For
			example, selecting a threshold of p = 0.01 (2 on -log10 scale) on the x-axis shows the
			proportion of observed genesets with p ≤ 0.01 on the y-axis. Since randomized genesets were
			used in these simulations, expected values are on the diagonal (dashed line).</p></div> <div style="margin-top: 100px"><h2 class="svelte-1vmxgqy">Description of the GOAT algorithm</h2> <p>In brief, the Geneset Ordinal Association Test (GOAT) is a parameter-free permutation-based
			algorithm for geneset enrichment analysis. It is easy to use via the online webtool or R
			package, computationally efficient and the resulting geneset p-values are well calibrated
			under the null hypothesis and invariant to geneset size. Application to various real-world
			proteomics and gene expression studies demonstrates that GOAT consistently identifies more
			significant Gene Ontology terms as compared to alternative methods.</p> <div><img src="goat_algorithm.png" alt="GOAT algorithm"/></div> <div style="padding: 0px 50px 0px 10px;"><ol><li class="svelte-1vmxgqy">Required input are a list of genes and their respective test statistics (p-value /
					effectsize), and a list of genesets obtained from GO or alternative resources.</li> <li class="svelte-1vmxgqy">Test statistics from the genelist are transformed to gene scores by rank(-pvalue)^2 or
					rank(effectsize)^2 depending on user-input, i.e. smaller p-values translate to higher gene
					scores. The result is a skewed gene score distribution.</li> <li class="svelte-1vmxgqy">For each geneset size N (number of genes), bootstrapping procedures generate a null
					distribution of geneset scores. This yields a skewed-normal distribution for small
					genesets and converges to a normal distribution for large genesets.</li> <li class="svelte-1vmxgqy">Geneset significance is determined for each geneset by comparing its score (mean of
					respective gene scores) against a null distribution of the same size (N).</li></ol></div></div>`;return{c(){m=g("meta"),_=G(),t=g("nav"),u=g("div"),f=g("a"),f.innerHTML=R,k=G(),o=g("a"),o.textContent=D,z=G(),l=g("a"),l.textContent=F,M=G(),r=g("a"),r.textContent=H,I=G(),d=g("a"),d.textContent=E,w=G(),y=g("div"),y.innerHTML=L,C=G(),x=g("div"),x.innerHTML=B,this.h()},l(e){const s=V("svelte-1b313dc",document.head);m=c(s,"META",{name:!0,content:!0}),s.forEach(h),_=O(e),t=c(e,"NAV",{});var p=S(t);u=c(p,"DIV",{style:!0});var N=S(u);f=c(N,"A",{href:!0,"data-svelte-h":!0}),T(f)!=="svelte-1kp1n6l"&&(f.innerHTML=R),N.forEach(h),k=O(p),o=c(p,"A",{href:!0,style:!0,"data-svelte-h":!0}),T(o)!=="svelte-9ooqgz"&&(o.textContent=D),z=O(p),l=c(p,"A",{href:!0,style:!0,"data-svelte-h":!0}),T(l)!=="svelte-10g8z77"&&(l.textContent=F),M=O(p),r=c(p,"A",{href:!0,style:!0,"data-svelte-h":!0}),T(r)!=="svelte-13lvetu"&&(r.textContent=H),I=O(p),d=c(p,"A",{href:!0,style:!0,"data-svelte-h":!0}),T(d)!=="svelte-1106bxz"&&(d.textContent=E),p.forEach(h),w=O(e),y=c(e,"DIV",{class:!0,"data-svelte-h":!0}),T(y)!=="svelte-p77g5c"&&(y.innerHTML=L),C=O(e),x=c(e,"DIV",{style:!0,"data-svelte-h":!0}),T(x)!=="svelte-1y2idh4"&&(x.innerHTML=B),this.h()},h(){document.title="GOAT: Geneset Ordinal Association Test",b(m,"name","description"),b(m,"content","Geneset enrichment analysis for Geneset Ontology (GO) or KEGG pathways using the GOAT algorithm webtool. Online data analysis for your preranked genelist from e.g. proteomics or bulk/scRNAseq gene expression studies"),b(f,"href","/"),v(f,"active",i(n[0],"github.io/goat")||i(n[0],"/index")),A(u,"padding","4px"),A(u,"margin-left","20px"),b(o,"href","/"),A(o,"margin-left","5px"),v(o,"active",i(n[0],"github.io/goat")||i(n[0],"/index")),b(l,"href","/goat"),A(l,"margin-left","40px"),v(l,"active",i(n[0],"/goat")),b(r,"href","/genemap"),A(r,"margin-left","40px"),v(r,"active",i(n[0],"/genemap")),b(d,"href","/docs"),A(d,"margin-left","40px"),v(d,"active",i(n[0],"/docs")),b(y,"class","divTitle svelte-1vmxgqy"),A(x,"background-color","white"),A(x,"padding","0px 50px 0px 50px")},m(e,s){a(document.head,m),q(e,_,s),q(e,t,s),a(t,u),a(u,f),a(t,k),a(t,o),a(t,z),a(t,l),a(t,M),a(t,r),a(t,I),a(t,d),q(e,w,s),q(e,y,s),q(e,C,s),q(e,x,s)},p(e,[s]){s&1&&v(f,"active",i(e[0],"github.io/goat")||i(e[0],"/index")),s&1&&v(o,"active",i(e[0],"github.io/goat")||i(e[0],"/index")),s&1&&v(l,"active",i(e[0],"/goat")),s&1&&v(r,"active",i(e[0],"/genemap")),s&1&&v(d,"active",i(e[0],"/docs"))},i:j,o:j,d(e){e&&(h(_),h(t),h(w),h(y),h(C),h(x)),h(m)}}}function W(n,m,_){let t;return $(n,Q,u=>_(0,t=u)),[t]}class te extends K{constructor(m){super(),J(this,m,W,U,P,{})}}export{te as component};
