import{s as P,f as o,a as A,A as S,g as r,d as l,c as G,h as B,B as O,j as y,C as q,k as b,x as i,i as _,y as D,z as V}from"../chunks/scheduler.7144be00.js";import{S as K,i as J}from"../chunks/index.3da70c46.js";/* empty css                    */import{p as Q}from"../chunks/stores.69e95f1a.js";import{e as s}from"../chunks/singletons.0b797c22.js";function U(f){let d,T,e,a,p,w=`<img src="${s+"/android-chrome-192x192.png"}" width="40" height="40" alt="GOAT"/>`,M,g,E="Home",I,c,F="GOAT online",R,h,H="gene ID mapping",z,m,$="Documentation",C,x,L=`<h1 style="" class="svelte-1vmxgqy">GOAT online geneset enrichment analysis</h1> <div><h3 class="svelte-1vmxgqy">Identify enriched genesets in a preranked genelist generated by e.g. proteomics or gene
			expression studies using the GOAT algorithm.</h3></div>`,k,v,N=`<div style="padding-top: 50px"><h2 class="svelte-1vmxgqy">The GOAT algorithm for geneset enrichment analysis</h2> <div><p>The GOAT algorithm has not been published yet but a preprint is available, please cite it
				when using the early-access version of GOAT;
				<br/> <i>Koopmans, F. (2023). GOAT: efficient and robust identification of geneset enrichment.</i> <br/> <a href="https://doi.org/10.1101/2023.12.10.570979" target="_blank" rel="nofollow">https://doi.org/10.1101/2023.12.10.570979</a></p></div> <div style="margin-top: 25px;">GOAT features:
			<ul class="svelte-1vmxgqy"><li class="svelte-1vmxgqy">Sensitive; more significant genesets as compared to other methods</li> <li class="svelte-1vmxgqy">Accurate; geneset p-values are accurate under the null hypothesis</li> <li class="svelte-1vmxgqy">Fast; completes in seconds</li> <li class="svelte-1vmxgqy">No arbitrary parameters (unlike ORA, no fiddle parameters for &quot;foreground genes&quot;)</li> <li class="svelte-1vmxgqy">Bootstrapping approach always uses appropriate &quot;background set&quot; (unlike ORA)</li> <li class="svelte-1vmxgqy">Available as a R package and through this website (<i>GOAT online</i> in above menu)</li></ul> <span><i>*ORA: overrepresentation analysis (i.e. classical Fisher-exact or hypergeometric test)</i></span></div></div> <div style="padding-top: 50px"><h2 class="svelte-1vmxgqy">GOAT finds more significant genesets</h2> <div>GOAT identifies more significant GO terms as compared to GSEA and ORA across 6 omics studies.</div> <div style="margin-top: 25px;"><img src="${s+"barplot_signif_count.svg"}" width="800" height="450" alt="barplot significant geneset counts"/></div> <p style="padding: 0px 50px 0px 20px;">Differential expression analysis results from each study (panels) were extracted from
			supplementary data and subjected to 5 approaches (colors) for identifying overrepresented GO
			terms. GO MF, GO CC and GO BP represent the respective Gene Ontology domains Molecular
			Functions, Cellular Components and Biological Processes. The x-axis shows the number of
			significant genesets identified by each method after Bonferroni adjustment at α = 0.05
			and adjustment for testing 3 sources of genesets.</p></div> <div style="padding-top: 50px"><h2 class="svelte-1vmxgqy">GOAT p-values are accurate</h2> <div>In simulations that generate 200 thousand random genelists and genesets, from small to large,
			we find that geneset p-values estimated by GOAT are accurate under the null hypothesis.</div> <div style="margin-top: 25px;"><img src="${s+"/null_simulations.svg"}" width="800" height="225" alt="GOAT p-values are accurate under the null hypothesis"/></div> <p style="padding: 0px 50px 0px 20px;">The y-axis shows the observed geneset p-value for a given p-value threshold on the x-axis
			(both on -log10 scale, p-values are shown as-is without multiple testing correction). For
			example, selecting a threshold of p = 0.01 (2 on -log10 scale) on the x-axis shows the
			proportion of observed genesets with p ≤ 0.01 on the y-axis. Since randomized genesets were
			used in these simulations, expected values are on the diagonal (dashed line).</p></div> <div style="margin-top: 100px"><h2 class="svelte-1vmxgqy">Description of the GOAT algorithm</h2> <p>In brief, the Geneset Ordinal Association Test (GOAT) is a parameter-free permutation-based
			algorithm for geneset enrichment analysis. It is easy to use via the online webtool or R
			package, computationally efficient and the resulting geneset p-values are well calibrated
			under the null hypothesis and invariant to geneset size. Application to various real-world
			proteomics and gene expression studies demonstrates that GOAT consistently identifies more
			significant Gene Ontology terms as compared to alternative methods.</p> <div><img src="${s+"goat_algorithm.png"}" alt="GOAT algorithm"/></div> <div style="padding: 0px 50px 0px 10px;"><ol><li class="svelte-1vmxgqy">Required input are a list of genes and their respective test statistics (p-value /
					effectsize), and a list of genesets obtained from GO or alternative resources.</li> <li class="svelte-1vmxgqy">Test statistics from the genelist are transformed to gene scores by rank(-pvalue)^2 or
					rank(effectsize)^2 depending on user-input, i.e. smaller p-values translate to higher gene
					scores. The result is a skewed gene score distribution.</li> <li class="svelte-1vmxgqy">For each geneset size N (number of genes), bootstrapping procedures generate a null
					distribution of geneset scores. This yields a skewed-normal distribution for small
					genesets and converges to a normal distribution for large genesets.</li> <li class="svelte-1vmxgqy">Geneset significance is determined for each geneset by comparing its score (mean of
					respective gene scores) against a null distribution of the same size (N).</li></ol></div></div>`;return{c(){d=o("meta"),T=A(),e=o("nav"),a=o("div"),p=o("a"),p.innerHTML=w,M=A(),g=o("a"),g.textContent=E,I=A(),c=o("a"),c.textContent=F,R=A(),h=o("a"),h.textContent=H,z=A(),m=o("a"),m.textContent=$,C=A(),x=o("div"),x.innerHTML=L,k=A(),v=o("div"),v.innerHTML=N,this.h()},l(t){const u=S("svelte-1b313dc",document.head);d=r(u,"META",{name:!0,content:!0}),u.forEach(l),T=G(t),e=r(t,"NAV",{});var n=B(e);a=r(n,"DIV",{style:!0});var j=B(a);p=r(j,"A",{href:!0,"data-svelte-h":!0}),O(p)!=="svelte-6nvu5r"&&(p.innerHTML=w),j.forEach(l),M=G(n),g=r(n,"A",{href:!0,style:!0,"data-svelte-h":!0}),O(g)!=="svelte-xsa54j"&&(g.textContent=E),I=G(n),c=r(n,"A",{href:!0,style:!0,"data-svelte-h":!0}),O(c)!=="svelte-o349n8"&&(c.textContent=F),R=G(n),h=r(n,"A",{href:!0,style:!0,"data-svelte-h":!0}),O(h)!=="svelte-sn5cge"&&(h.textContent=H),z=G(n),m=r(n,"A",{href:!0,style:!0,"data-svelte-h":!0}),O(m)!=="svelte-1dikf5g"&&(m.textContent=$),n.forEach(l),C=G(t),x=r(t,"DIV",{class:!0,"data-svelte-h":!0}),O(x)!=="svelte-p77g5c"&&(x.innerHTML=L),k=G(t),v=r(t,"DIV",{style:!0,"data-svelte-h":!0}),O(v)!=="svelte-vhtur9"&&(v.innerHTML=N),this.h()},h(){document.title="GOAT: Geneset Ordinal Association Test",y(d,"name","description"),y(d,"content","Geneset enrichment analysis for Geneset Ontology (GO) or KEGG pathways using the GOAT algorithm webtool. Online data analysis for your preranked genelist from e.g. proteomics or bulk/scRNAseq gene expression studies"),y(p,"href",s+"/"),q(p,"active",f[0]==="home"),b(a,"padding","4px"),b(a,"margin-left","20px"),y(g,"href",s+"/"),b(g,"margin-left","5px"),q(g,"active",f[0]==="home"),y(c,"href",s+"/goat"),b(c,"margin-left","40px"),q(c,"active",f[0]==="goat"),y(h,"href",s+"/genemap"),b(h,"margin-left","40px"),q(h,"active",f[0]==="genemap"),y(m,"href",s+"/docs"),b(m,"margin-left","40px"),q(m,"active",f[0]==="docs"),y(x,"class","divTitle svelte-1vmxgqy"),b(v,"background-color","white"),b(v,"padding","0px 50px 0px 50px")},m(t,u){i(document.head,d),_(t,T,u),_(t,e,u),i(e,a),i(a,p),i(e,M),i(e,g),i(e,I),i(e,c),i(e,R),i(e,h),i(e,z),i(e,m),_(t,C,u),_(t,x,u),_(t,k,u),_(t,v,u)},p:D,i:D,o:D,d(t){t&&(l(T),l(e),l(C),l(x),l(k),l(v)),l(d)}}}function W(f,d,T){let e;V(f,Q,w=>T(1,e=w));const a=e&&e.url?e.url.pathname.replace(".html",""):"/";return[a===s+"/goat"&&"goat"||a===s+"/docs"&&"docs"||a===s+"/genemap"&&"genemap"||"home"]}class se extends K{constructor(d){super(),J(this,d,W,U,P,{})}}export{se as component};
