import{s as Me,f as a,a as r,l as $,A as He,g as o,d as t,c as u,h as v,m as J,B as c,j as p,C as Ie,k as d,x as i,i as n,y as re,z as Ce}from"../chunks/scheduler.cd38125d.js";import{S as Ee,i as De}from"../chunks/index.a6a2bf8f.js";/* empty css                    */import{p as ze}from"../chunks/stores.3ed8a801.js";function Le(g){let f,L,l,b,y,w,ue,Q,I,X,Z,M,ee,te,H,ie,se,x,ae,j,C,pe="Documentation",P,E,de=`<h3>Citation</h3> <div><p>The GOAT algorithm has not been published yet but a preprint is available, please cite it when
			using the early-access version of GOAT;
			<br/> <i>Koopmans, F. (2023). GOAT: efficient and robust identification of geneset enrichment.</i> <br/><a href="https://doi.org/10.1101/2023.12.10.570979" target="_blank" rel="nofollow">https://doi.org/10.1101/2023.12.10.570979</a></p> <p>A ready-made M&amp;M text describing your GOAT analysis is included with this tool&#39;s results
			(c.f. the log file contained in the output ZIP-file).</p></div>`,R,T,he='<h3>How do I use this tool?</h3> <div><div>brief overview of workflow (more elaborate docs coming soon)</div> <div><ol><li class="svelte-pooqga">input data: genelist (your dataset, prepared as specified in the section below)</li> <li class="svelte-pooqga">select genesets (e.g. GO or SynGO, provided by GOAT online)</li> <li class="svelte-pooqga">settings (e.g. for multiple testing correction)</li> <li class="svelte-pooqga">run the analysis by clicking &quot;START&quot;</li> <li class="svelte-pooqga">download results; Excel table with all genesets and M&amp;M for your paper</li> <li class="svelte-pooqga">use interactive data analysis tools to visualize/interpret results</li></ol></div></div> <br/> <b>Can I use the GOAT algorithm programmatically?</b> <div style="margin-top: 5px;">Yes! We also provide a R package; <a href="https://github.com/ftwkoopmans/goat" target="_blank" rel="nofollow">via this link</a></div>',W,G,ce=`<h3>Expected file format for the input genelist:</h3> <div><ul><li class="svelte-pooqga">File format: either CSV, TSV or Excel (.xlsx file, data on the first sheet). Note that for
				Excel, the old .xls is not supported, only the newer .xlsx format.</li> <li class="svelte-pooqga">Required columns (column names must match exactly)</li> <ol><li class="svelte-pooqga"><b>gene</b>: Human Entrez (NCBI) gene IDs (integer values)</li> <li class="svelte-pooqga"><b>symbol</b>: gene symbol (at least 2 characters)</li> <li class="svelte-pooqga"><b>effectsize</b>: effectsize or log2 foldchange (numeric/decimal values)</li> <li class="svelte-pooqga"><b>pvalue</b>: gene p-values (numeric/decimal values). Preferably, use the un-adjusted
					p-values because this information is used for sorting and after adjustment, there
					typically are many more ties (e.g. p-values set to 1) which in turn causes a loss of
					information</li> <li class="svelte-pooqga"><b>signif</b>: was the adjusted p-value significant? This column should contain boolean
					values, i.e. (true and false, or 0 and 1). Here one should use the adjusted p-values!
					While this information is NOT used by the GOAT algorithm to identify significant genesets,
					flagging proteins that are significant in your genelist/dataset will yield useful
					information in downstream interpretation of your data. For example, in the GOAT result
					tables you can see for each geneset how many (and which) significant genes are present.
					<br/></li></ol> <li class="svelte-pooqga">Missing/empty values are not allowed, except for the &#39;gene&#39; column; rows where this column
				is empty will be ignored/skipped.</li> <li class="svelte-pooqga">Duplicate entries for genes, i.e. multiple rows with the same value in the &#39;gene&#39; column,
				are reduced to only 1 row; whichever has the lowest p-value (and if there is no p-value
				column, whichever row has the highest absolute effectsize).</li> <li class="svelte-pooqga">While you can upload genelists that only have either an &#39;effectsize&#39; or &#39;pvalue&#39; column, it
				is recommended to always include both if this information is available in your dataset
				because both sources of information can be used when sorting/ranking the genelist (e.g. when
				sorting by p-value, effectsizes can be used to break ties especially for proteins with
				p-value=1).</li></ul></div> <div style="margin-bottom: 20px;"><b>importantly</b>, only Human Entrez (NCBI) gene IDs are supported for now (i.e. values in the
		&#39;gene&#39; column).
		<br/>
		We provide a gene ID mapping tool (available through the menu on top of this screen) to easily add
		Entrez gene IDs to your genelists in case these only contain gene symbols.</div> <div style="margin-bottom: 10px;">Example genelist: <a href="data/example_genelist_Wingo_pmid32424284.xlsx">click here to download</a>
		the Wingo et al. 2020 dataset in Excel format (<a href="https://pubmed.ncbi.nlm.nih.gov/32424284" target="_blank" rel="nofollow">PMID:32424284</a>) . You may use this as an example for preparing your genelist in a format compatible with
		GOAT.</div>`,B,_,ge=`<h3>Using custom genesets</h3> <div>Genesets prepared in the generic GMT file format can be imported. For example, KEGG pathways in
		GMT format can be downloaded <a href="https://www.gsea-msigdb.org/gsea/msigdb/collections.jsp" target="_blank" rel="nofollow">via this link</a>. Navigate to the &quot;KEGG_LEGACY subset of CP&quot; section and select &quot;NCBI (Entrez) Gene IDs&quot;. This
		should yield a file with a name similar to &quot;c2.cp.kegg.v2023.1.Hs.entrez.gmt&quot; (depending on the
		version). GOAT online only works when input genelists and genesets use Entrez Gene identifiers,
		so it is crucial to select the appropriate gene format when preparing/downloading genesets in
		GMT format.
		<br/>You may use the genesets in this file using the &quot;upload GMT file&quot; button in the &quot;Genesets&quot;
		section of the GOAT online tool.</div>`,S,q,me=`<h3>Frequently Asked Questions (FAQ)</h3> <p><i>How do we future-proof this tool? e.g. prevent it from growing stale/outdated and keep it
			online ?</i> <br/>The website is implemented as a fully &quot;static&quot; HTML + Javascript website, so hosting the
		website is trivial; we use GitHub Pages to host it. As long as GitHub is online, so is this
		tool.
		<br/>To ensure access to recent Gene Ontology database data, we are currently setting up an
		automated workflow for importing the latest GO database N times per year using GitHub Actions
		(i.e. won&#39;t rely on us manually updating this website).
		<br/>Further, you can always import any geneset collection in GMT format into this webtool.</p> <p><i>Does the webtool yield the same geneset p-values as the R package?</i> <br/>Yes. We validated that the geneset p-values computed by GOAT online and the GOAT R package
		are the same across all 7 &#39;omics datasets that are described in the GOAT manuscript.</p>`,Y,h,V,fe="Description of the GOAT algorithm",oe,N,ve=`In brief, the Geneset Ordinal Association Test (GOAT) is a parameter-free permutation-based
		algorithm for geneset enrichment analysis. It is easy to use via the online webtool or R
		package, computationally efficient and the resulting geneset p-values are well calibrated under
		the null hypothesis and invariant to geneset size. Application to various real-world proteomics
		and gene expression studies demonstrates that GOAT consistently identifies more significant Gene
		Ontology terms as compared to alternative methods.`,ne,F,D,be,le,z,ye=`<ol><li class="svelte-pooqga">Required input are a list of genes and their respective test statistics (p-value /
				effectsize), and a list of genesets obtained from GO or alternative resources.</li> <li class="svelte-pooqga">Test statistics from the genelist are transformed to gene scores by rank(-pvalue)^2 or
				rank(effectsize)^2 depending on user-input, i.e. smaller p-values translate to higher gene
				scores. The result is a skewed gene score distribution.</li> <li class="svelte-pooqga">For each geneset size N (number of genes), bootstrapping procedures generate a null
				distribution of geneset scores. This yields a skewed-normal distribution for small genesets
				and converges to a normal distribution for large genesets.</li> <li class="svelte-pooqga">Geneset significance is determined for each geneset by comparing its score (mean of
				respective gene scores) against a null distribution of the same size (N).</li></ol>`,K,A,we='<h3>Privacy</h3> <ul><li class="svelte-pooqga">all analyses are performed locally on your computer using client-side Javascript code</li> <li class="svelte-pooqga">your genelist and all analyses thereof remain private, your data does not leave your computer</li> <li class="svelte-pooqga">we do count the number of times this tool is used, anonymously, to gauge its popularity</li></ul>',U,O,xe=`<h3>Logo</h3> <p>the GOAT logo shown on this website is borrowed from the open source Noto Emoji Font version
		14.0</p>`;return{c(){f=a("meta"),L=r(),l=a("nav"),b=a("div"),y=a("a"),w=a("img"),Q=r(),I=a("a"),X=$("Home"),Z=r(),M=a("a"),ee=$("GOAT online"),te=r(),H=a("a"),ie=$("gene ID mapping"),se=r(),x=a("a"),ae=$("Documentation"),j=r(),C=a("h1"),C.textContent=pe,P=r(),E=a("div"),E.innerHTML=de,R=r(),T=a("div"),T.innerHTML=he,W=r(),G=a("div"),G.innerHTML=ce,B=r(),_=a("div"),_.innerHTML=ge,S=r(),q=a("div"),q.innerHTML=me,Y=r(),h=a("div"),V=a("h3"),V.textContent=fe,oe=r(),N=a("p"),N.textContent=ve,ne=r(),F=a("div"),D=a("img"),le=r(),z=a("div"),z.innerHTML=ye,K=r(),A=a("div"),A.innerHTML=we,U=r(),O=a("div"),O.innerHTML=xe,this.h()},l(e){const s=He("svelte-1m51m41",document.head);f=o(s,"META",{name:!0,content:!0}),s.forEach(t),L=u(e),l=o(e,"NAV",{});var m=v(l);b=o(m,"DIV",{style:!0});var Te=v(b);y=o(Te,"A",{href:!0});var Ge=v(y);w=o(Ge,"IMG",{src:!0,width:!0,height:!0,alt:!0}),Ge.forEach(t),Te.forEach(t),Q=u(m),I=o(m,"A",{href:!0,style:!0});var _e=v(I);X=J(_e,"Home"),_e.forEach(t),Z=u(m),M=o(m,"A",{href:!0,style:!0});var qe=v(M);ee=J(qe,"GOAT online"),qe.forEach(t),te=u(m),H=o(m,"A",{href:!0,style:!0});var Ae=v(H);ie=J(Ae,"gene ID mapping"),Ae.forEach(t),se=u(m),x=o(m,"A",{href:!0,style:!0,class:!0});var Oe=v(x);ae=J(Oe,"Documentation"),Oe.forEach(t),m.forEach(t),j=u(e),C=o(e,"H1",{"data-svelte-h":!0}),c(C)!=="svelte-fjilyk"&&(C.textContent=pe),P=u(e),E=o(e,"DIV",{"data-svelte-h":!0}),c(E)!=="svelte-1vomu16"&&(E.innerHTML=de),R=u(e),T=o(e,"DIV",{style:!0,"data-svelte-h":!0}),c(T)!=="svelte-1tidcjw"&&(T.innerHTML=he),W=u(e),G=o(e,"DIV",{style:!0,"data-svelte-h":!0}),c(G)!=="svelte-1tx9yrt"&&(G.innerHTML=ce),B=u(e),_=o(e,"DIV",{style:!0,"data-svelte-h":!0}),c(_)!=="svelte-ktgcbp"&&(_.innerHTML=ge),S=u(e),q=o(e,"DIV",{style:!0,"data-svelte-h":!0}),c(q)!=="svelte-14e1wl6"&&(q.innerHTML=me),Y=u(e),h=o(e,"DIV",{style:!0});var k=v(h);V=o(k,"H3",{"data-svelte-h":!0}),c(V)!=="svelte-1xztreg"&&(V.textContent=fe),oe=u(k),N=o(k,"P",{"data-svelte-h":!0}),c(N)!=="svelte-1zbiyw"&&(N.textContent=ve),ne=u(k),F=o(k,"DIV",{});var ke=v(F);D=o(ke,"IMG",{src:!0,alt:!0,width:!0}),ke.forEach(t),le=u(k),z=o(k,"DIV",{style:!0,"data-svelte-h":!0}),c(z)!=="svelte-xqe9tv"&&(z.innerHTML=ye),k.forEach(t),K=u(e),A=o(e,"DIV",{style:!0,"data-svelte-h":!0}),c(A)!=="svelte-1k4jaeb"&&(A.innerHTML=we),U=u(e),O=o(e,"DIV",{style:!0,"data-svelte-h":!0}),c(O)!=="svelte-8ovfie"&&(O.innerHTML=xe),this.h()},h(){document.title="GOAT: Geneset Ordinal Association Test - documentation",p(f,"name","description"),p(f,"content","Geneset enrichment analysis for Geneset Ontology (GO) or KEGG pathways using the GOAT algorithm webtool. Online data analysis for your preranked genelist from e.g. proteomics or bulk/scRNAseq gene expression studies"),Ie(w.src,ue=g[0]+"android-chrome-192x192.png")||p(w,"src",ue),p(w,"width","40"),p(w,"height","40"),p(w,"alt","GOAT"),p(y,"href",g[0]),d(b,"padding","4px"),d(b,"margin-left","20px"),p(I,"href",g[0]),d(I,"margin-left","5px"),p(M,"href",g[0]+"goat"),d(M,"margin-left","40px"),p(H,"href",g[0]+"genemap"),d(H,"margin-left","40px"),p(x,"href",g[0]+"docs"),d(x,"margin-left","40px"),p(x,"class","active"),d(T,"margin-top","50px"),d(G,"margin-top","50px"),d(_,"margin-top","50px"),d(q,"margin-top","50px"),Ie(D.src,be=g[0]+"goat_algorithm.png")||p(D,"src",be),p(D,"alt","GOAT algorithm"),p(D,"width","900"),d(z,"padding","0px 50px 0px 10px"),d(h,"margin-top","50px"),d(A,"margin-top","50px"),d(O,"margin-top","50px")},m(e,s){i(document.head,f),n(e,L,s),n(e,l,s),i(l,b),i(b,y),i(y,w),i(l,Q),i(l,I),i(I,X),i(l,Z),i(l,M),i(M,ee),i(l,te),i(l,H),i(H,ie),i(l,se),i(l,x),i(x,ae),n(e,j,s),n(e,C,s),n(e,P,s),n(e,E,s),n(e,R,s),n(e,T,s),n(e,W,s),n(e,G,s),n(e,B,s),n(e,_,s),n(e,S,s),n(e,q,s),n(e,Y,s),n(e,h,s),i(h,V),i(h,oe),i(h,N),i(h,ne),i(h,F),i(F,D),i(h,le),i(h,z),n(e,K,s),n(e,A,s),n(e,U,s),n(e,O,s)},p:re,i:re,o:re,d(e){e&&(t(L),t(l),t(j),t(C),t(P),t(E),t(R),t(T),t(W),t(G),t(B),t(_),t(S),t(q),t(Y),t(h),t(K),t(A),t(U),t(O)),t(f)}}}function Ve(g,f,L){let l;return Ce(g,ze,y=>L(1,l=y)),[!!l&&l.url.hostname.endsWith("github.io")&&"/goat/"||"/"]}class Re extends Ee{constructor(f){super(),De(this,f,Ve,Le,Me,{})}}export{Re as component};
